pdns/syncres.hh

```diff
@@ -114,6 +114,38 @@ public:
     DNSName d_name;
     bool d_rdForward{false};
 
+    bool operator==(const AuthDomain& rhs) const
```

Comment:
IMO both these functions do not need to be inline. The code is only used for debugging and this header file is already too big...